### DISCUSSION TITLE
Problem Verbindung Node-RED zu PixelIt

### DISCUSSION THREAD

**Discussion by guybaust** [pixelit-project/PixelIt#86] [Q&A]
Hallo,
Ich habe mir ebenfalls eine PixelIt gebastelt, funktioniert auch soweit, das heißt ich sehe die Uhr auf der Matrix.
Auf einen Raspberry Pi 4 habe ich Home Assistant und Node-RED installiert und den Einsteiger Flow geladen.
Wenn ich dann allerdings den Flow auf meine PixelIt übertragen will krieg ich die Meldung "Cannot reach the PixelIt via http API".
In dem Button "Core" habe ich die IP Adresse der PixelIt eingegeben (WiFi 192.168.173.xxx). Der Raspi hat die Adresse (Kabel 192.168.1.xxx. Kann dies das Problem sein (verschiedene Adressen) oder muß ich noch irgendwas installieren? Beide Adressen kann ich von meinem Mac anpingen.
Besten Dank im voraus
Guy

**Comment by o0shojo0o** [comment]
Hi, 
ja die Netzt müssen untereinander geroutet werden, sonst können diese ja nicht miteinander kommunizieren!

Ohne diesen Blog Post genauer  gelesen zu haben könnte der dir helfen.
https://administrator.de/tutorial/routing-von-2-und-mehr-ip-netzen-mit-windows-linux-und-router-56073.html

**Reply by guybaust** [reply]
Danke für die schnelle Antwort.
Leider nicht. Beide haben jetzt eine 192.168.1.xxx Adresse aber immer noch das gleiche Problem. Nur die Farbe der Uhr wechselt von weiß auf rot wenn ich auf deploy klicke. MQTT muss ich ja nicht installieren bzw  auf der Matrix konfigurieren?

**Reply by o0shojo0o** [reply]
Nein, MQTT wird nicht benötigt und ist rein optional. 
Aber das die Farbe der Uhr wechselt ist echt komisch, dass hat der Flow nicht mit drinnen. 

Du nutzte den Flow oder? 
https://docs.bastelbunker.de/pixelit/tools.html#einsteiger-flow

**Reply by guybaust** [reply]
Ja, genau dieser. Wenn ich den Flow importiere sieht er aber nicht so aus wie der Screenshot in deinem Link.

Bin aber wieder ein kleines Stück weiter. Ich habe hier noch einen anderen Raspi mit PiHole drauf. Wenn ich den deaktiviere funktioniert die Kommunikation mit dem PixelIt. Warum das so ist? Keine Ahnung, ich dachte immer PiHole wäre nur für das Netzwerk von außen zuständig und nicht für innen. Aber egal, Hauptsache es geht jetzt. 

Allerdings scheint der Flow nicht so richtig zu laufen bei mir denn er wechselt die Farbe auf Rot und abwechselnd erscheint die Uhr und eine Grafik wo links zwei senkrechte Striche sind und rechts steht die Zahl 22, weiter nichts.

Ich benutze als LED Displays 4 stück von den CJMCU 8x8 weil das andere nicht lieferbar war und einen Clone ebenfalls von AZDelivery. Kann man irgendwo noch weitere Flows mit Grafiken downloaden die fix und fertig sind zum aufspielen?

Auf jeden Fall , vielen Dank bis hierhin für deine Hilfe.
Guy

**Comment by o0shojo0o** [comment]
Ja, das Bild eine eine alte Version vom Einsteiger Flow.
Aber deine Fehlerbeschreibung sieht fast nach einer Inkompatibilität der LED Displays aus :( 
Ich selbst habe hier aber keine CJMCU liegen daher kann ich es nicht validieren.. 

Hier ein Flow wo Du einfach die ID im Node "Bitmap Animation" nach belieben wechseln kannst 
(http://\<IP-Pixel It\>/#/gallery)
```json 
[
    {
        "id": "35ecb3fe.70a82c",
        "type": "Core",
        "z": "6cea977d.420938",
        "name": "Core",
        "ip": "192.168.0.137",
        "masterTopic": "",
        "x": 1470,
        "y": 180,
        "wires": [
            []
        ]
    },
    {
        "id": "2ae8103e.2bfa8",
        "type": "Alert Screen",
        "z": "6cea977d.420938",
        "name": "Alert Screen",
        "x": 1330,
        "y": 180,
        "wires": [
            [
                "35ecb3fe.70a82c"
            ]
        ]
    },
    {
        "id": "2ae3a14.25d9d5e",
        "type": "inject",
        "z": "6cea977d.420938",
        "name": "Push",
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": "0",
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 510,
        "y": 180,
        "wires": [
            [
                "a8d0668.e8c4098"
            ]
        ]
    },
    {
        "id": "8ad9260b.cf8098",
        "type": "Switch Animation",
        "z": "6cea977d.420938",
        "name": "Switch Animation",
        "aktiv": "true",
        "animation": "fade",
        "x": 1030,
        "y": 180,
        "wires": [
            [
                "b3ad9478.01eb48"
            ]
        ]
    },
    {
        "id": "b3ad9478.01eb48",
        "type": "Text",
        "z": "6cea977d.420938",
        "name": "Text",
        "intextString": "Pixel!!",
        "inbigFont": "false",
        "inscrollText": "true",
        "inscrollTextDelay": "0",
        "incenterText": "false",
        "inposX": "8",
        "inposY": "1",
        "incolorR": "255",
        "incolorG": "255",
        "incolorB": "255",
        "x": 1190,
        "y": 180,
        "wires": [
            [
                "2ae8103e.2bfa8"
            ]
        ]
    },
    {
        "id": "a8d0668.e8c4098",
        "type": "Screen Init",
        "z": "6cea977d.420938",
        "name": "Screen Init",
        "inscreenName": "bra",
        "induration": "10",
        "x": 650,
        "y": 180,
        "wires": [
            [
                "b3f96a15.ff4fb8"
            ]
        ]
    },
    {
        "id": "b3f96a15.ff4fb8",
        "type": "Bitmap Animation",
        "z": "6cea977d.420938",
        "name": "Bitmap Animation",
        "indata": "4",
        "inanimationDelay": "100",
        "inrubberbanding": "false",
        "inlimitLoops": "0",
        "x": 830,
        "y": 180,
        "wires": [
            [
                "8ad9260b.cf8098"
            ]
        ]
    }
]
```

**Reply by guybaust** [reply]
Hallo,
Ich raffs einfach  nicht.
Heute bekam ich noch ein weiteres LED Display. Scheint das gleiche zu sein wie das in der Teileliste verlinkt ist allerdings von einem anderen Hersteller. Allerdings verhält es sich genau so wie das andere Display. Verbindung steht, der Einsteiger Flow scheint auch übertragen zu werden allerdings wird nichts korrekt angezeigt. Die LEDs werden zwar nicht mehr rot, das kommt nur wenn man mit dem D1 verbunden ist und in der Test Area dann unter Brightness den Wert auf 1 stellt. Ich hatte den auf 1 gestellt weils so hell war. Läßt man ihn auf 127 ist die Uhr auch korrekt. Animation wechselt nur zwischen Uhr an einer Grafik wo links ein P steht und dann rechts die Zahl 22.

![IMG_7910](https://user-images.githubusercontent.com/24720536/143690135-443bb52f-018b-4b65-abbe-45534bebb893.JPG)


![IMG_7909](https://user-images.githubusercontent.com/24720536/143690140-2e3bbde4-cc23-4eab-bc4a-993b58cc522d.JPG)

Außerdem verstehe ich nicht was ich mit dem Flow (ist es überhaupt ein Flow den ich in Node-RED importieren muß? scheint nicht so weil übertragen wird da nichts) machen muß.

Kannst du mir bitte erklären was du mit dem Satz

"Hier ein Flow wo Du einfach die ID im Node "Bitmap Animation" nach belieben wechseln kannst
(http://<IP-Pixel It>/#/gallery"

genau meinst.

(http://<IP-Pixel It>/#/gallery" ist soweit klar,
bei mir http://192.168.1.130/#/gallery und dann...?

entschuldige bitte meine blöden Fragen aber vielleicht ist da noch irgendwas was ich nicht richtig eingestellt oder verstanden habe, z.B was hat es mit dem roten Text rechts im Bild auf sich?

<img width="627" alt="Bildschirmfoto 2021-11-27 um 18 09 34" src="https://user-images.githubusercontent.com/24720536/143690654-f66792de-78d2-40a5-a0a1-d9c15a30baf4.png">

Fragen über Fragen
Guy

**Reply by o0shojo0o** [reply]
Jetzt komme ich bei dir nicht mehr mit, 
Die beiden Bilder von dir Zeigen doch den Flow korrekt an!

Das erste Bild von Dir zeigt den `Github PixelIt Stars` Zweig im Flow.
Das zweite Bild Zeigt `Internal Pixel It Clock [internal_clock]` Zweig im Flow.

Der `Xmas Counter !!! Will only be visible when only 24 days left !!!` Wird erst sichtbar, wenn nur noch 24 Tage übrig sind zu Weihnachten.
Und die `Alert Message` wird erst Sichtbar wenn Du sie Triggers. (Vorne auf den Push Button drückts).

In der Gallery kannst du die Bitmaps Direkt zum ansehen auch auf die Pixel It senden indem auf dem `rot markierten` Symbol klickst.
Die `grün markierte` Zahl ist die BMP ID die in den NodeRed Flows nutzen kannst, so wir das Bild vom Webservice abgerufen.
![image](https://user-images.githubusercontent.com/19333515/143696357-52311cb9-b456-4c12-b65d-a3c0640ecb6d.png)

Den Flow den ich dir ein den Post exportiert habe kannst due einfach importiren so wie du es schon mit dem Einsteiger Flow gemacht hast.

**Reply by guybaust** [reply]
Hallo Dennis,
OK, ich dachte in dem Flow würde "mehr" passieren. Das mit dem Push wußte ich nicht. Sorry dafür.
Als Mac User bin vielleicht zu sehr verwöhnt daß immer alles auf Anhieb gleich funktioniert.
Nur als Info: mit Safari kann ich das PixelIt nicht ansteuern auf dem Mac (Browserfenster bleibt leer). Mit Firefox gehts.

Eigentlich wollte ja nur ein paar Animationen und die Uhr abwechselnd auf dem Display darstellen.
Aber das ist jetzt für mich nicht so einfach zu realisieren.

Ich baue mir jetzt noch ein schönes Gehäuse aus Holz und werde das ganze dann als Uhr benutzen.
Danke
Guy
